optuna/integration/tensorflow.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import absolute_import
+
+import optuna
+
+
+try:
+    import tensorflow as tf
+    from tensorflow.train import SessionRunHook
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # PruningHook is disabled because TensorFlow is not available.
+    _available = False
+    SessionRunHook = object
+
+
+class TensorFlowPruningHook(SessionRunHook):
+    """TensorFlow SessionRunHook to prune umpromising trials.
+
+    Example:
+
+        Add a pruning SessionRunHook for a TensorFlow's Estimator.
+
+        .. code::
+
+                optuna_pruning_hook = OptunaPruningHook(
+                    trial=trial,
+                    estimator=clf,
+                    metric="accuracy",
+                    is_higher_better=True,
+                    run_every_steps=10,
+                )
+                tf.estimator.train_and_evaluate(
+                    clf,
+                    tf.estimator.TrainSpec(input_fn=train_input_fn, max_steps=500, hooks=hooks),
+                    eval_spec
+                )
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of
+            the objective function.
+        estimator:
+            A :estimator which you will use.
+        metric:
+            An evaluation metric for pruning, e.g., ``accuracy`` and ``loss``.
+        is_higher_better:
+           It should be True if you use a metric to be maximize such as ``loss``.
+        run_every_steps:
+           An interval to watch the summary file.
+    """
+    def __init__(self, trial, estimator, metric, is_higher_better, run_every_steps):
```

Comment:
Could you add a type hint to `TensorFlowPruningHook.__init__`? To keep compatibility with Python 2.7, we use type comments as follows:
```
    def __init__(
            self,
            trial,  # type: optuna.trial.Trial
            ....
    ):
        # type: (...) -> None

        self.trial = trial
```
Please refer to [`Study.__init__`](https://github.com/pfnet/optuna/blob/master/optuna/study.py#L58) as an example.

Reply:
I added a type hint as below:

```python
# type: (optuna.trial.Trial, tf.estimator.Estimator, str, bool, int) -> None
```